File: problems/003-nth-fibonacci/rust/brute-force.rs
use crate::json::JVal;

fn fib(n: i64) -> i64 {
    if n < 2 { n } else { fib(n - 1) + fib(n - 2) }
}

pub fn solve(c: &JVal) -> JVal {
    let n = c.get("n").as_num() as i64;
    // Obvious approach: the naive recursive definition. O(phi^n) time, O(n) stack.
    JVal::Num(fib(n) as f64)
}

File: problems/003-nth-fibonacci/java/Brute-force.java
import java.util.*;
// Brute force: the naive recursive definition. Exponential O(phi^n) time,
// linear call-stack depth. The obvious statement of the problem, and the
// baseline the O(n) Clean/Optimized improve on. Non-public (file is Brute-force.java).
class BruteForce implements Solution {
    public Object solve(Map<String, Object> c) {
        long n = ((Number) c.get("n")).longValue();
        return fib(n);
    }
    static long fib(long n) {
        return n < 2 ? n : fib(n - 1) + fib(n - 2);
    }
}
